components/list/list-item-drag-drop-mixin.js

```diff
@@ -380,11 +468,26 @@ export const ListItemDragDropMixin = superclass => class extends superclass {
 	}
 
 	_onDragEnd(e) {
+
 		const dragState = getDragState();
 		this.dragging = false;
-		if (dragState.shouldDrop(e.timeStamp)) {
-			this._annoucePositionChange(dragState.dragTargetKey, dragState.dropTargetKey, dragState.dropLocation);
+
+		if (e.dataTransfer.dropEffect !== 'none' && dragState.shouldDrop(e.timeStamp)) {
```

Comment:
`dropEffect === 'none'` when the user hits Escape while dragging. (`dragend` event always fires to we need to check this before dispatching our custom move event).